### DISCUSSION TITLE
I want to separate SELECT query to my team and other users - Row Level Security

### DISCUSSION THREAD

**Discussion by davut** [supabase/supabase#4551] [Questions] [answered]
I have a table of tools. 

I want to list all active tools (active == true) to all subscribed users.

But for my team I want to list all tools ( by all I mean all active and not active tools) 

So here are my policies

1 - `active = true AND right(auth.email(), 12) != '@example.com'` - get all active tools who does not have @example.com
2 - `right(auth.email(), 12) = '@example.com'` - Get all tools for teams that has @example.com email

Thanks

**Comment by GaryAustin1** [comment, marked as answer]
I would think you could simplify to just USING (active = true OR right(auth.email(),12) = '@example.com') .

**Reply by davut** [reply]
That makes sense but even only `right(auth.email(),12) = '@example.com'` returning me empty array. How can I check what auth.email() is returning?

**Reply by GaryAustin1** [reply]
Are you sure you are logged in when you make the call?  
There is now logging for the database under settings.  I've not played with that yet for debugging policies but I'm hopeful that will help in situations like this to see internal workings.   Otherwise you might throw an error with message to get any info.  
EDIT how do you know it is returning empty array?

**Reply by davut** [reply]
Yes I am, under the identity I can see `my-email@example.com` as logged in.

**Reply by davut** [reply]
Because It is not throwing any errors

**Reply by GaryAustin1** [reply]
It might help if you show your exact policy as your info shows 1,2 with description of what you want to do.  I've not used auth.email() but it is shown in many examples so I assume works.

**Reply by davut** [reply]
Ok so I figured it out. I was requesting in getServerSideProps and I was not able to read RLS table in getServerSideProps. 



```Javascript
supabase.auth.session = () => ({
    access_token: context.req.cookies['sb:token']
});
```
Adding this before my query solved the issue.

Also mentioned [here](https://github.com/supabase/supabase/issues/1735#issuecomment-922284089) 

@GaryAustin1 Thank you for your help
